### ISSUE TITLE
View alias's in prometheus metric es_indices_mappings_field_count

### ISSUE THREAD

**Issue by jonny-wg2** [braedon/prometheus-es-exporter#69]
I would like to be able to filter on alias's for index's. 

For example, our es cluster stores 60 indexes. Generally it is `logstash-yyyy-mm-dd`. 

The issue I have is, because the index date changes, I am unable to create a prometheus metric out of `es_indices_mappings_field_count` to monitor the count of fields in today's metric. 

https://stackoverflow.com/questions/58734793/use-today-date-in-label-values-in-prometheus-alerting-rules/58745416#58745416

**Comment by braedon** [comment]
Hi @jonny-wg2, I can add metrics for index aliases, allowing you to filter other metrics based on those aliases.

e.g. assuming your alias was `logstash-today`:
```
sum(es_indices_mappings_field_count and on (index) es_indices_aliases_alias{alias="logstash-today"}) by (index)
```
Note that when the index/alias rolls over at the end of the day the old alias metric will persist for a few minutes even though the exporter is no longer exporting it. This will temporarily result in two metrics for the same alias, e.g.:
```
es_indices_aliases_alias{index="logstash-2020-06-30", alias="logstash-today"} 1.0
es_indices_aliases_alias{index="logstash-2020-07-01", alias="logstash-today"} 1.0
```
This will in turn result in two results for the above query - one for each of the indices - for this period.

Would that work for you?

**Comment by jonny-wg2** [comment]
Ahh that would be awesome! I am ok with have a duplicate metric for some time.

Question, would the exporter be able to add the `logstash-today` alias to the metric? This would be beneficial as it's one less cronjob I would need to run.

**Comment by braedon** [comment]
> Question, would the exporter be able to add the `logstash-today` alias to the metric? This would be beneficial as it's one less cronjob I would need to run.

Sorry, not sure what you mean by this. Could you elaborate?

**Comment by jonny-wg2** [comment]
Sorry I'll try to elaborate. 

Will the `alias="logastash-today"` be populated by the es-exporter for today's index, or do I need to add logic elseware to add this alias outside? It would be great if the exporter could add this alias to the correct index.

**Comment by braedon** [comment]
Ahh, I see. Unfortunately, no, I'd prefer to keep the exporter free of that kind of feature. The `es_indices_aliases_alias` metrics would be populated by parsing the `/_alias` endpoint, so any alias you wanted to use would need to be added to Elasticsearch and maintained separately.

**Comment by jonny-wg2** [comment]
Ah ok, because I know that indices can support multiple alias's so it shouldn't be a conflict to user's setup if this was the case. But that ok. I'll just build a cronjob to add this to our ES cluster every 24 hours.

**Comment by braedon** [comment]
Released in [0.11.0](https://github.com/braedon/prometheus-es-exporter/releases/tag/0.11.0)
